projects/kit/components/input-date/multi-date/input-date-multi.component.ts

```diff
@@ -0,0 +1,347 @@
+import {maskitoDateOptionsGenerator} from '@maskito/kit';
+import {
+    AbstractTuiMultipleControl,
+    AbstractTuiValueTransformer,
+    ALWAYS_FALSE_HANDLER,
+    changeDateSeparator,
+    TUI_DATE_FORMAT,
+    TUI_DATE_SEPARATOR,
+    TUI_IS_MOBILE,
+    tuiAsControl,
+    tuiAsFocusableItemAccessor,
+    TuiBooleanHandler,
+    tuiDateClamp,
+    TuiDateMode,
+    TuiDay,
+    TuiFocusableElementAccessor,
+    tuiIsString,
+    TuiMapper,
+    TuiMonth,
+} from '@taiga-ui/cdk';
+import {
+    TUI_DEFAULT_MARKER_HANDLER,
+    TUI_TEXTFIELD_SIZE,
+    TuiDialogService,
+    TuiMarkerHandler,
+    TuiPrimitiveTextfieldComponent,
+    TuiSizeL,
+    TuiSizeS,
+    TuiTextfieldSizeDirective,
+    TuiWithOptionalMinMax,
+} from '@taiga-ui/core';
+import {TuiNamedDay, TuiStringifiableItem} from '@taiga-ui/kit/classes';
+import {TuiInputTagComponent} from '@taiga-ui/kit/components/input-tag';
+import {
+    TUI_DATE_TEXTS,
+    TUI_DATE_VALUE_TRANSFORMER,
+    TUI_DONE_WORD,
+    TUI_INPUT_DATE_OPTIONS,
+    TUI_MOBILE_CALENDAR,
+    tuiDateStreamWithTransformer,
+    TuiInputDateOptions,
+} from '@taiga-ui/kit/tokens';
+import {tuiImmutableUpdateInputDateMulti} from '@taiga-ui/kit/utils';
+import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+import {Observable} from 'rxjs';
+import {map, takeUntil} from 'rxjs/operators';
+
+@Component({
+    selector: 'tui-input-date[multiple]',
+    templateUrl: './input-date-multi.template.html',
+    styleUrls: ['../input-date.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        tuiAsFocusableItemAccessor(TuiInputDateMultiComponent),
+        tuiAsControl(TuiInputDateMultiComponent),
+        tuiDateStreamWithTransformer(TUI_DATE_VALUE_TRANSFORMER),
+    ],
+})
+export class TuiInputDateMultiComponent
+    extends AbstractTuiMultipleControl<TuiDay>
+    implements TuiWithOptionalMinMax<TuiDay>, TuiFocusableElementAccessor
+{
+    @ViewChild(TuiPrimitiveTextfieldComponent)
+    private readonly textfield?: TuiPrimitiveTextfieldComponent;
+
+    @ViewChild(TuiInputTagComponent)
+    private readonly inputTag?: TuiInputTagComponent;
+
+    private month: TuiMonth | null = null;
+
+    @Input()
+    min: TuiDay | null = this.options.min;
+
+    @Input()
+    max: TuiDay | null = this.options.max;
+
+    @Input()
+    disabledItemHandler: TuiBooleanHandler<TuiDay> = ALWAYS_FALSE_HANDLER;
+
+    @Input()
+    markerHandler: TuiMarkerHandler = TUI_DEFAULT_MARKER_HANDLER;
+
+    @Input()
+    items: readonly TuiNamedDay[] = [];
+
+    @Input()
+    defaultActiveYearMonth = TuiMonth.currentLocal();
+
+    @Input()
+    expandable = false;
+
+    @Input()
+    inputHidden = false;
+
+    @Input()
+    @HostBinding('class._editable')
+    editable = true;
+
+    @Input()
+    search: string | null = '';
+
+    @Input()
+    placeholder = '';
+
+    @Input()
+    rows = Infinity;
+
+    @Input()
+    maskitoOptions: MaskitoOptions = maskitoDateOptionsGenerator({
```

Comment:
Let's not expose it as `@Input` for now.

Reply:
fixed